### DISCUSSION TITLE
running EDIPIC-2d

### DISCUSSION THREAD

**Discussion by ausefi** [PrincetonUniversity/EDIPIC-2D#15] [General]
hi, I installed EDIPIC-2D on ubuntu linux successfully, when I am trying to run the example of it with this command on intel core 2 du , I recieved this error. how can I run the code in single computer with core 2 du :
`ali@ali/usr/local/projects/Example_beam$ mpirun -np 1 ./edipic2d >& output.log & tail -f output.log`
[1] 5925
*** The MPI_Comm_f2c() function was called before MPI_INIT was invoked.
*** This is disallowed by the MPI standard.
*** Your MPI job will now abort.
[ali:05930] Local abort before MPI_INIT completed completed successfully, but am not able to aggregate error messages, and not able to guarantee that all other processes were killed!

Primary job terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.

mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

Process name: [[45880,1],0]
Exit code: 1

**Comment by stephethier** [comment]
Hi there,
Thanks for trying out EDIPIC-2D. The number of MPI tasks that you use on your "mpirun" line, which is 1 here, needs to correspond to the number of tasks specified in the "init_configuration.dat" file. These are the lines that you need to change:

```
---ddd---------- number of blocks (processes) along the X (horizontal) direction 
     4       <----- Change this number to 1
---ddd---------- number of blocks (processes) along the Y (vertical) direction 
     8       <----- Change this number to 1 
```
You will also need to change the number of cells per block (process) if you want the calculation to be the same.

```
---ddd---------- number of cells along the X-direction in a block
    60      <----- This number becomes 4*60 = 240
---ddd---------- number of cells along the Y-direction in a block 
    60      <----- This number becomes  8*60 = 480
```
I'm actually not sure if EDIPIC-2D can run with a single MPI process so give it a try and see what happens.
